scripts/wi-setup.sh

```diff
@@ -0,0 +1,106 @@
+    if ! az account show > /dev/null 2>&1; then
+        echo -e "\e[31mPlease login to Azure CLI using 'az login'\e[0m"
+        exit 1
+    fi
+}
+
+create_azure_blob_storage_account() {
+    export RESOURCE_GROUP="oidc-issuer"
+    export LOCATION="westus2"
+    if [ $(az group exists --name "${RESOURCE_GROUP}") == 'false' ]; then
+        echo -e "\e[32mCreating resource group '${RESOURCE_GROUP}' in '${LOCATION}'\e[0m"
+        az group create --name "${RESOURCE_GROUP}" --location "${LOCATION}"
+    fi
+
+    export AZURE_STORAGE_ACCOUNT="oidcissuer$(openssl rand -hex 4)"
+    export AZURE_STORAGE_CONTAINER="oidc-test"
+    if ! az storage account show --name "${AZURE_STORAGE_ACCOUNT}" --resource-group "${RESOURCE_GROUP}" > /dev/null 2>&1; then
+        echo -e "\e[32mCreating storage account '${AZURE_STORAGE_ACCOUNT}' in '${RESOURCE_GROUP}'\e[0m"
+        az storage account create --resource-group "${RESOURCE_GROUP}" --name "${AZURE_STORAGE_ACCOUNT}" --allow-blob-public-access true
+    fi
+
+    if ! az storage container show --name "${AZURE_STORAGE_CONTAINER}" --account-name "${AZURE_STORAGE_ACCOUNT}" > /dev/null 2>&1; then
+        echo -e "\e[32mCreating storage container '${AZURE_STORAGE_CONTAINER}' in '${AZURE_STORAGE_ACCOUNT}'\e[0m"
+        az storage container create --name "${AZURE_STORAGE_CONTAINER}" --public-access container
+    fi
+
+    export SERVICE_ACCOUNT_ISSUER="https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_STORAGE_CONTAINER}/"
+}
+
+upload_openid_docs(){
+    echo -e "\e[32mGetting public signing key from the cluster\e[0m"
+    docker exec ${KIND_CLUSTER_NAME}-control-plane cat /etc/kubernetes/pki/sa.pub > "${SCRIPT_PATH}/sa.pub"
+
+    cat <<EOF > "${SCRIPT_PATH}/openid-configuration.json"
+{
+  "issuer": "${SERVICE_ACCOUNT_ISSUER}",
+  "jwks_uri": "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_STORAGE_CONTAINER}/openid/v1/jwks",
+  "response_types_supported": [
+    "id_token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256"
+  ]
+}
+EOF
+
+    echo -e "\e[32mUploading openid-configuration document to '${AZURE_STORAGE_ACCOUNT}' storage account\e[0m"
+    upload_to_blob "${AZURE_STORAGE_CONTAINER}" "${SCRIPT_PATH}/openid-configuration.json" ".well-known/openid-configuration"
+
+    azwi jwks --public-keys "${SCRIPT_PATH}/sa.pub" --output-file "${SCRIPT_PATH}/jwks.json"
+    echo -e "\e[32mUploading jwks document to '${AZURE_STORAGE_ACCOUNT}' storage account\e[0m"
+    upload_to_blob "${AZURE_STORAGE_CONTAINER}" "${SCRIPT_PATH}/jwks.json" "openid/v1/jwks"
+}
+
+upload_to_blob() {
+    local container_name=$1
+    local file_path=$2
+    local blob_name=$3
+
+    echo -e "\e[32mUploading ${file_path} to '${AZURE_STORAGE_ACCOUNT}' storage account\e[0m"
+    az storage blob upload \
+        --container-name "${container_name}" \
+        --file "${file_path}" \
+        --name "${blob_name}"
+}
+
+create_kind_cluster() {
```

Comment:
where is this going to be used? I think we discussed not having kind in here so users can choose their environment

Reply:
+1, let's avoid references to kind in this script if possible

Reply:
This is that new script specifically for the Kind.

Reply:
we discussed today to make `kind` optional here (can be a flag in the script) and assuming a cluster is pre-configured